front/lib/resources/data_source_view_resource.ts

```diff
@@ -204,12 +204,15 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
   ) {
     const dataSourceViews = await this.baseFetch(
       auth,
-      fetchDataSourceViewOptions
+      fetchDataSourceViewOptions,
+      {
+        where: {
+          workspaceId: auth.getNonNullableWorkspace().id,
+        },
+      }
     );
 
-    return dataSourceViews.filter(
-      (dsv) => auth.isAdmin() || auth.hasPermission([dsv.vault.acl()], "read")
-    );
+    return dataSourceViews.filter((dsv) => dsv.canList(auth));
```

Comment:
I understand your intent but the change is not a 1-1 change i believe. You may break something.